File: 프로그래머스/2/42586. 기능개발/기능개발.swift
import Foundation

func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    let remainProgress = progresses.map { 100 - $0 }
    var remainDays = zip(remainProgress, speeds).map { (($0 - 1) / $1) + 1}
    var result = [Int]()
    
    while !remainDays.isEmpty {
        let todayTask = remainDays.removeFirst()
        var todayDone = 1
        
        while !remainDays.isEmpty && todayTask >= remainDays[0] {
            todayDone += 1
            remainDays.removeFirst()
        }
        result.append(todayDone)
    }
    
    return result
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    remain = [100 - x for x in progresses]
    dueDate = []
    answer = []

    for (r, s) in zip(remain, speeds):
        if r % s == 0:
            dueDate.append(r // s)
        else:
            dueDate.append(r // s + 1)

    while dueDate:
        todayTask = dueDate.pop(0)
        todayDone = 1
        while True:
            if dueDate and todayTask >= dueDate[0]:
                dueDate.pop(0)
                todayDone += 1
            else:
                break
        answer.append(todayDone)
    return answer